lib/abr/simple_abr_manager.js

```diff
@@ -243,6 +264,30 @@ shaka.abr.SimpleAbrManager = class {
   }
 
 
+  /**
+   * @override
+   * @export
+   */
+  setMediaElement(mediaElement) {
+    this.mediaElement_ = mediaElement;
+    if (this.resizeObserver_) {
+      this.resizeObserver_.disconnect();
+      this.resizeObserver_ = null;
+    }
+    if (this.mediaElement_ && 'ResizeObserver' in window) {
+      this.resizeObserver_ = new ResizeObserver(() => {
+        if (this.config_.qualityRestrictionsElementSize) {
+          const chosenVariant = this.chooseVariant();
```

Comment:
How rapidly can the ResizeObserver fire?  If I'm dragging the corner of the browser window slowly, and the video element expands with the window, will this choose variants rapidly (once a second or more)?

If the answer is that it fires rapidly, we may need to add a Timer object to batch up decisions until the size settles.  Thankfully, this is pretty easy to do.  See `keyStatusTimer_` in `DrmEngine` if you need an example.

Reply:
You're right, it can fired many times per second in that case... It's necessary add a timer.

Reply:
Done!